FirebaseDatabase/Sources/Utilities/FNextPushId.m

```diff
@@ -59,4 +65,43 @@ + (NSString *)get:(NSTimeInterval)currentTime {
     return [NSString stringWithString:id];
 }
 
++ (NSString *)nextAfter:(NSString *_Nonnull)key {
+    NSInteger keyAsInt;
+    if ([FUtilities tryParseStringToInt:key asInt:&keyAsInt]) {
+        return [FUtilities
+            ieee754StringForNumber:[NSNumber numberWithInteger:(keyAsInt + 1)]];
+    }
+    NSString *charFormat = @"%C";
+    NSMutableString *next = [NSMutableString stringWithString:key];
+    if ([next length] < MAX_KEY_LEN) {
+        [next insertString:MIN_PUSH_CHAR atIndex:[key length]];
+        return next;
+    }
+    unsigned long i = [next length] - 1;
+
+    while (i >= 0 && [[next substringWithRange:NSMakeRange(i, i + 1)]
+                         isEqualToString:MAX_PUSH_CHAR]) {
+        i--;
+    }
+
+    // `nextAfter` was called on the largest possible key, so return the
+    // maxName, which sorts larger than all keys.
+    if (i == -1) {
+        return [FUtilities maxName];
+    }
+
+    NSString *source =
+        [NSString stringWithFormat:charFormat, [next characterAtIndex:i]];
+    NSString *sourcePlusOne = [NSString
+        stringWithFormat:charFormat,
+                         [PUSH_CHARS
+                             characterAtIndex:[PUSH_CHARS rangeOfString:source]
+                                                  .location +
+                                              1]];
```

Comment:
This is everything that is wrong with Objective-C in 7 lines, but I can't think of a better way. It might help to create an extra variable to store `[PUSH_CHARS rangeOfString:source].location` in, but I am not sure this is worth saving.